moveit_setup_assistant/moveit_setup_controllers/src/moveit_controllers_config.cpp

```diff
@@ -0,0 +1,230 @@
+  else if (joints_node.IsDefined())
+  {
+    control_setting.joints_.push_back(joints_node.as<std::string>());
+  }
+  if (control_setting.joints_.empty())
+  {
+    RCLCPP_ERROR_STREAM((*logger_), "Couldn't parse joints for controller " << name << " in moveit_controllers.yaml");
+    return false;
+  }
+  // All required fields were parsed correctly
+  controllers_.push_back(control_setting);
+  return true;
+}
+
+// ******************************************************************************************
+// Output controllers config files
+// ******************************************************************************************
+bool MoveItControllersConfig::GeneratedControllersConfig::writeYaml(YAML::Emitter& emitter)
+{
+  emitter << YAML::Comment("MoveIt uses this configuration for controller management");
+  emitter << YAML::Newline;
+  emitter << YAML::BeginMap;
+  {
+    // TODO: Output possible trajectory_execution parameters including...
+    //       * allowed_execution_duration_scaling: 1.2
+    //       * allowed_goal_duration_margin: 0.5
+    //       * allowed_start_tolerance: 0.01
+
+    emitter << YAML::Key << "moveit_controller_manager" << YAML::Value
+            << "moveit_simple_controller_manager/MoveItSimpleControllerManager";
+    emitter << YAML::Newline;
+    emitter << YAML::Newline;
+
+    emitter << YAML::Key << "moveit_simple_controller_manager" << YAML::Value;
+    emitter << YAML::BeginMap;
+    {
+      emitter << YAML::Key << "controller_names";
+      emitter << YAML::Value;
+      emitter << YAML::BeginSeq;
+      for (const ControllerInfo& ci : parent_.controllers_)
+      {
+        emitter << ci.name_;
+      }
+      emitter << YAML::EndSeq;
+      emitter << YAML::Newline;
+      emitter << YAML::Newline;
+
+      for (const auto& controller : parent_.controllers_)
+      {
+        emitter << YAML::Key << controller.name_;
+        emitter << YAML::Value;
+        emitter << YAML::BeginMap;
+        {
+          emitter << YAML::Key << "type" << YAML::Value << controller.type_;
+          // Write joints
+          emitter << YAML::Key << "joints";
+          emitter << YAML::Value;
+          if (controller.joints_.size() != 1)
+          {
+            emitter << YAML::BeginSeq;
+
+            // Iterate through the joints
+            for (const std::string& joint : controller.joints_)
+            {
+              emitter << joint;
+            }
+            emitter << YAML::EndSeq;
+          }
+          else
+          {
+            emitter << YAML::BeginMap;
+            emitter << controller.joints_[0];
+            emitter << YAML::EndMap;
+          }
+          // Depending on the controller type, fill the required data
+          if (controller.type_ == "FollowJointTrajectory")
+          {
+            emitter << YAML::Key << "action_ns" << YAML::Value << "follow_joint_trajectory";
+            emitter << YAML::Key << "default" << YAML::Value << "true";
+          }
+          else
```

Comment:
I _believe_ you could safely drop this, or make it an error. All of the MoveIt ControllerManager stuff only works with FollowJointTrajectory, so there would be no reason to configure anything else.

@JafarAbdi @schornakj  do you agree?

```suggestion
          else
          {
            // Throw error or something
            RCLCPP_WARN_STREAM((*logger_),
                           "MoveIt controllers only work with FollowJointTrajectory-type controllers.");
            return false;
          }
```